src/v/kafka/protocol/batch_reader.cc

```diff
@@ -0,0 +1,141 @@
+// Copyright 2020 Vectorized, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "kafka/protocol/batch_reader.h"
+
+#include "kafka/protocol/kafka_batch_adapter.h"
+#include "model/record.h"
+#include "model/timeout_clock.h"
+#include "vlog.h"
+
+#include <seastar/core/coroutine.hh>
+#include <seastar/core/loop.hh>
+#include <seastar/core/std-coroutine.hh>
+
+namespace kafka {
+
+namespace {
+
+/// \brief A subset of the model::record_batch_header
+///
+/// Just enough to delineate record_batch within consumer_records.
+struct record_batch_info {
+    model::offset base_offset{};
+    int32_t record_count{};
+    int32_t size_bytes{};
```

Comment:
is this 0? why not set it to 0. what's the default initializer of an int.

Reply:
i know sizet / unsigned are specified. are ints well specified in every compiler?

Reply:
This is zero invalidation via value initialization via default member initialization.

Happy make it explicit for the benefit of doubt.